File: Problems 0000-0099/06. Zigzag Conversion/solution.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1 || numRows >= s.length()) {
            return s;
        }
        vector<string> rows(numRows, "");
        int currRow = 0;
        int step = 1;
        for (int i = 0; i < s.length(); i++) {
            rows[currRow] += s[i];
            if (currRow == 0) {
                step = 1;
            } else if (currRow == numRows-1) {
                step = -1;
            }
            currRow += step;
        }

        string res = "";
        for (const string& row : rows) {
            res += row;
        }
        return res;
    }
};

File: Problems 0000-0099/06. Zigzag Conversion/solution.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or numRows >= len(s):
            return s

        rows = [[] for _ in range(numRows)]

        step = 1
        currRow = 0
        for i in range(len(s)):
            rows[currRow].append(s[i])
            if currRow == 0:
                step = 1
            elif currRow == numRows - 1:
                step = -1
            currRow += step
        
        return "".join(["".join(row) for row in rows])
        